src/codeu/chat/client/simplegui/UserPanel.java

```diff
@@ -158,8 +158,19 @@ public void actionPerformed(ActionEvent e) {
       public void actionPerformed(ActionEvent e) {
         if (userList.getSelectedIndex() != -1) {
           final String data = userList.getSelectedValue();
-          clientContext.user.signInUser(data);
-          userSignedInLabel.setText("Hello " + data);
+          //Ask user for password
+          final String userPassword = (String) JOptionPane.showInputDialog(
+            UserPanel.this, "Enter " + data + "'s password:", "Enter Password", JOptionPane.PLAIN_MESSAGE,
+            null, null, "");
+          //check password to make sure it is correct and check it against the user's password
+          if(userPassword.equals("password")){ //access userByName, get User, access the user's password (String)
```

Comment:
I'm not sure I understand this line. Why are we checking that the user's password is "password"? Does this mean the user's password must be "password"?